CMakeLists.txt

```diff
@@ -177,7 +177,7 @@ elseif(MZ_ZLIB)
         clone_repo(ZLIB https://github.com/madler/zlib)
 
         # Don't automatically add all targets to the solution
-        add_subdirectory(${ZLIB_SOURCE_DIR} ${ZLIB_BINARY_DIR} EXCLUDE_FROM_ALL)
+        add_subdirectory(${ZLIB_SOURCE_DIR} ${ZLIB_SOURCE_DIR} EXCLUDE_FROM_ALL)
```

Comment:
No need to specify `${ZLIB_SOURCE_DIR}` twice.

Reply:
Do you mean `add_subdirectory(${ZLIB_SOURCE_DIR} EXCLUDE_FROM_ALL)` would do the job?

Reply:
Yes, if you change it I think we can merge it.

Reply:
Just tested it. It does not work with `add_subdirectory(${ZLIB_SOURCE_DIR} EXCLUDE_FROM_ALL)`, it really needs to be `add_subdirectory(${ZLIB_SOURCE_DIR} ${ZLIB_SOURCE_DIR} EXCLUDE_FROM_ALL)`

Reply:
Thanks!

Reply:
> . The FetchContent_Populate call would set it as zlib_BINARY_DIR ...

Are you sure this is correct? The `name` argument passed to `clone_repo` is always uppercase.

Reply:
From the cmake documentation at https://cmake.org/cmake/help/latest/module/FetchContent.html

FetchContent_Populate() will set three variables in the scope of the caller; <lcName>_POPULATED, <lcName>_SOURCE_DIR and <lcName>_BINARY_DIR, where <lcName> is the lowercased <name>.

Reply:
Actually docs for cmake shows it forces all lowercase..

> FetchContent_Populate() will set three variables in the scope of the caller; <lcName>_POPULATED, <lcName>_SOURCE_DIR and <lcName>_BINARY_DIR, where <lcName> is the lowercased <name>..

Reply:
Yeah, initially I changed all the code in the clone_repo to adjust for this, but in the end i still needed the one line fix in this PR. So I went with the minimal fix.